lib/puppet/provider/common.rb

```diff
@@ -25,15 +25,15 @@ def pretty(arg)
 end
 
 # Removes quotes from nil and false values
-def data_parse(data = {})
+def data_parse(data = {}, client = nil)
```

Comment:
`data_parse` calls another method  named `uri_parsing`, which make some rest calls.
Then, I need to provide a client for this method.
I believe the responsability of the client creation should not be from the `uri_parsing` method, as it is for now, it should be made only by the Puppet::OneViewResource and Puppet::ImageStreamerResource classes.
 
As far as I can see, that method doesn't know which type of client should be instantiated: OneViewSDK::Client or OneViewSDK::ImageStreamer::Client